libbeat/publisher/beat/pipeline.go

```diff
@@ -57,6 +65,29 @@ type ClientConfig struct {
 	ACKLastEvent func(Event)
 }
 
+// ClientEventer provides access to internal client events.
+type ClientEventer interface {
+	Closing() // Closing indicates the client is being shutdown next
```

Comment:
Kind of expected these methods to return bool. But will probably see later how they are used.